src/main/java/org/apache/maven/plugin/compiler/AbstractCompilerMojo.java

```diff
@@ -703,8 +703,39 @@ protected final Optional<Path> getModuleDeclaration(final Set<File> sourceFiles)
     private boolean targetOrReleaseSet;
 
     @Override
-    @SuppressWarnings("checkstyle:MethodLength")
     public void execute() throws MojoExecutionException, CompilationFailureException {
+        try {
+            executeReal();
+        } finally {
+            addGeneratedSourcesToProject();
+        }
```

Comment:
I'm not sure if finally is needed ... if something wrong in first method whole Maven build will be broken anymore

Reply:
Yes and no: I agree that the `try / finally` is not required for most circumstances (regular `mvn clean compile`, etc.). However, maven also has the `--fail-at-end` and `--fail-never` flags. In this case, the try finally could make sense.

So, just to be on the safe side to not change the original behavior, I would keep the `try / finally`. However, I am also open to removing it, if you would prefer it that way.

Reply:
ok, can stays

Reply:
Try-finally is necessary, because there are several `return` points in the method and doing a finally block is the easiest way to ensure this work gets done, no matter how these `return`s are shuffled around in the future. It actually has nothing to do with failure.